parsl/executors/workqueue/executor.py

```diff
@@ -321,6 +321,32 @@ def submit(self, func, resource_specification, *args, **kwargs):
         kwargs : dict
             Keyword arguments to the Parsl app
         """
+        cores = None
+        memory = None
+        disk = None
+        if resource_specification and isinstance(resource_specification, dict):
+            logger.debug("Got resource specification: {}".format(resource_specification))
+
+            acceptable_resource_types = ['cores', 'memory', 'disk']
+            keys = list(resource_specification.keys())
+            if len(keys) != 3:
+                logger.error("Task resource specification requires "
+                             "three resources to be specified simultaneously: cores, memory, and disk")
+                raise ExecutorError(self, "Task resource specification requires "
+                                          "three resources to be specified simultaneously: cores, memory, and disk, "
+                                          "and only takes these three resource types.")
+
+            if not all(k.lower() in acceptable_resource_types for k in keys):
```

Comment:
is it right to have `.lower()` here? what happens one of the keys is "CORES"?

Reply:
The `lower()` here is meant to allow users to not worry about the case? so if a user does something like `parsl_resource_specification={'Cores', 'MEMORY', 'Disk'}`, that should also work.

Reply:
I missed that check in the later code. I will fix that